logic/src/commonMain/kotlin/com/wire/kalium/logic/data/message/PersistMessageUseCase.kt

```diff
@@ -12,22 +16,34 @@ interface PersistMessageUseCase {
     suspend operator fun invoke(message: Message.Standalone): Either<CoreFailure, Unit>
 }
 
+@OptIn(DelicateKaliumApi::class)
 internal class PersistMessageUseCaseImpl(
     private val messageRepository: MessageRepository,
-    private val selfUser: UserId
+    private val userRepository: UserRepository
 ) : PersistMessageUseCase {
     override suspend operator fun invoke(message: Message.Standalone): Either<CoreFailure, Unit> {
-        val isMyMessage = message.senderUserId == selfUser
-        @OptIn(DelicateKaliumApi::class)
+        val (updateConversationNotificationsDate, isMyMessage) = userRepository.getSelfUser()?.let {
+            message.shouldUpdateConversationNotificationDate(it) to message.isMyMessage(it.id)
+        } ?: (false to false)
         return messageRepository
             .persistMessage(
                 message = message,
                 updateConversationReadDate = isMyMessage,
                 updateConversationModifiedDate = message.content.shouldUpdateConversationOrder(),
-                updateConversationNotificationsDate = isMyMessage
+                updateConversationNotificationsDate
             )
     }
 
+    private fun Message.shouldUpdateConversationNotificationDate(selfUser: SelfUser) =
+        when (selfUser.availabilityStatus) {
+            UserAvailabilityStatus.AWAY -> this.isMyMessage(selfUser.id)
+            UserAvailabilityStatus.BUSY -> true // todo: OR conversationMutedStatus == MutedConversationStatus.OnlyMentionsAndRepliesAllowed
+            UserAvailabilityStatus.NONE -> true
+            UserAvailabilityStatus.AVAILABLE -> true
+        }
```

Comment:
are you sure about this `shouldUpdateConversationNotificationDate`?
i think it should be like: 

```
when (selfUser.availabilityStatus) {
    UserAvailabilityStatus.AWAY -> true
    UserAvailabilityStatus.BUSY -> this.isMyMessage(selfUser.id) // todo: OR conversationMutedStatus == MutedConversationStatus.OnlyMentionsAndRepliesAllowed
    else -> this.isMyMessage(selfUser.id)
}
```

Reply:
Correct, thanks for the catch👨‍🍳